ouroboros-consensus/test-consensus/Test/Consensus/Node.hs

```diff
@@ -132,35 +131,38 @@ test_checkProtocolMagicId_empty = res @?= Left e
     e = CorruptDbMarker fullPath
 
 {-------------------------------------------------------------------------------
-  lockDbMarkerFile
+  lockDb
 -------------------------------------------------------------------------------}
 
-test_lockDbMarkerFile :: Assertion
-test_lockDbMarkerFile = withTempDir $ \dbPath -> do
-    let mount = MountPoint dbPath
-    -- Create the DB marker file
-    checkDbMarker (ioHasFS mount) mountPoint expectedProtocolMagicId >>=
-      (@?= Right ())
+test_lockDb :: Assertion
+test_lockDb = withTempDir $ \dbPath -> do
+    let hasFS     = ioHasFS $ MountPoint dbPath
+        withLock  = withLockDB hasFS dbPath
+        touchLock = withLock $ return ()
 
     -- Lock it once
-    withRegistry $ \registry -> do
-      tryL (lockDbMarkerFile registry dbPath) >>=
+    tryL touchLock >>=
         (@?= (Right ()))
 
-      -- Try to lock it again. This should fail.
-      tryL (lockDbMarkerFile registry dbPath) >>=
-        (@?= (Left (DbLocked (dbPath </> T.unpack dbMarkerFile))))
-
-    -- Unlock it now by closing the 'ResourceRegistry'.
-    -- We must be able to lock it again now.
-    withRegistry $ \registry ->
-      tryL (lockDbMarkerFile registry dbPath) >>=
+    -- Raise an exception. The lock should get cleaned.
+    _ <- tryT (withLock $ throwM TestException)
+    tryL touchLock >>=
         (@?= (Right ()))
-    -- And finally unlock it.
+
+    -- Try to lock it twice. This should fail after 2sec.
+    -- This test succeeds, but may leave a forgotten lock to the file
+    -- tryL (withLock touchLock) >>=
+    --     (@?= (Left (DbLocked (dbPath </> T.unpack dbLockFile))))
```

Comment:
This test is commented out

Reply:
Yes, as I explain above the test `This test succeeds, but may leave a forgotten lock to the file` and this lock is only cleaned when all tests finish and the process dies. So probably we shouldn't include it?

Reply:
> Yes, as I explain above the test `This test succeeds, but may leave a forgotten lock to the file` and this lock is only cleaned when all tests finish and the process dies. So probably we shouldn't include it?

But now we are not even testing whether the locking actually works. 

> this lock is only cleaned when all tests finish and the process dies

That's fine, or what's wrong with that?